addons/mail/models/mail_compose_message.py

```diff
@@ -643,17 +666,101 @@ def _gc_lost_attachments(self):
             ('write_date', '<', limit_date)]
         ).unlink()
 
+    @api.autovacuum
+    def _gc_not_scheduled(self):
+        """ Garbage collect composers that are not scheduled (drafts that were not scheduled nor
+        sent, and sent composers) """
+        limit_date = fields.Datetime.subtract(fields.Datetime.now(), days=1)
+        self.search([('state', '!=', 'scheduled'), ('write_date', '<', limit_date)]).unlink()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        composers = super().create(vals_list)
+        # link attachments to composer (they were created with res_id=0)
+        for composer in composers:
+            if attachments := composer.attachment_ids:
+                attachments.filtered(
+                    lambda a: a.res_model == self._name and not a.res_id
+                ).write({'res_id': composer.id})
+        return composers
+
+    def write(self, vals):
+        if any(composer.state == 'sent' for composer in self):
+            raise UserError(_("Sent composers cannot be modified"))
+
+        res = super().write(vals)
+
+        # (re)schedule cron trigger if needed
+        trigger_datetimes = set()
+        if scheduled_date := vals.get('scheduled_date'):
```

Comment:
I was going to say: only scheduled composers should have a scheduled_date, then remembered it was also used due to templates, ended up scratching my head. As stated on task, I feel like this field has two usage that are not clearly split. Shouldn't we have two fields, or shouldn't we dive more into scheduled code for post / mailings ?